### PR TITLE
[M] ENT-5370: Migrate oauth_spec to Java

### PR DESCRIPTION
**Pull request by bakajstep**
- Added support for OAuth1 API client
- Migrate oauth_spec.rb
- Delete oauth_spec.rb

### CODE REVIEW

**Comment by bakajstep** [comment]
serialization_spec.rb is an old code that we don't need to migrate.

File: spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/ApiClients.java [hunk 1]
```diff
@@ -129,6 +129,43 @@ public static ApiClient bearerToken(String token) {
...
+     */
+    public static ApiClient oauthUser(String oauthConsumer, String oauthSecret, String username) {
+        return new ApiClient(CLIENT_FACTORY.createOauthClient(oauthConsumer, oauthSecret)
+            .addDefaultHeader("cp-user", username));
+    }
+
+    /**
+     * Returns a client to make API calls with, that uses an OAuth consumer authentication.
+     *
+     * @param oauthConsumer an OAuth consumer
```
Comment (Ceiu) [inline spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/ApiClients.java:159]: The term "consumer" is overloaded in this context (OAuth and Candlepin both have their own definitions of a consumer), so clarification is needed to differentiate them.

File: spec-tests/src/test/java/org/candlepin/spec/auth/OauthSpecTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Test
+    void shouldReturnsAUnauthorizedIfOauthSecretDoesNotMatch() {
+        ApiClient oauth = ApiClients.oauth(OAUTH_CONSUMER, "badsecret");
+        assertUnauthorized(() -> oauth.users().listUsers());
+    }
+
+    @Test
+    void shouldLetsACallerActAsAUser() {
```
Comment (Ceiu) [inline spec-tests/src/test/java/org/candlepin/spec/auth/OauthSpecTest.java:55]: This method, and the one following it, should read "shouldLetACaller..." -- the "Lets" should not be pluralized.

File: spec-tests/src/test/java/org/candlepin/spec/auth/OauthSpecTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Test
+    void shouldLetsACallerActAsAConsumer() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ApiClient oauthConsumer = ApiClients.oauthConsumer(
+            OAUTH_CONSUMER, OAUTH_SECRET, consumer.getUuid());
+        ConsumerDTO consumerFromServer = oauthConsumer.consumers().getConsumer(consumer.getUuid());
+        assertThat(consumerFromServer.getId())
```
Comment (Ceiu) [inline spec-tests/src/test/java/org/candlepin/spec/auth/OauthSpecTest.java:73]: Add a null check and return check here:

```java
assertThat(consumerFromServer)
  .isNotNull()
  .returns(consumer.getId(), ConsumerDTO::getId);
```

**Review by Ceiu — CHANGES_REQUESTED** [review]

File: spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/ApiClients.java (hunk 1 above)
Comment (bakajstep) [reply spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/ApiClients.java:173]: is that enough, or should I add something else I'm not sure about the acting like sentence

File: spec-tests/src/test/java/org/candlepin/spec/auth/OauthSpecTest.java (hunk 2 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/auth/OauthSpecTest.java]: done

File: spec-tests/src/test/java/org/candlepin/spec/auth/OauthSpecTest.java (hunk 3 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/auth/OauthSpecTest.java]: done

File: spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/ApiClients.java (hunk 1 above)
Comment (Ceiu) [reply spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/ApiClients.java:173]: The "acting like" sentences are extraneous and somewhat confusing, but the addition of "Candlepin" on the param is sufficient here.

**Review by Ceiu — APPROVED** [review]

**Pull request merged**